examples/dnsdoctor/dnsdoctor.lua

```diff
@@ -0,0 +1,80 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- Filter DNS packets based on a blocklist
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local function nop() end
+
+local function get_domain(skb, off)
+	local name = ""
+	local i = 0
+
+	repeat
+		local d = skb:getbyte(off + i)
+		i = i + 1
+	until d == 0
+
+	return name .. skb:getstring(off, i-1), i
+end
+
+local function dnsdoctor_tg(skb, par, userdata)
+	local sep = userdata:find(" ")
+	local target_dns = userdata:sub(1, sep - 1)
+	local dst_ip, target_ip = string.unpack("I4I4", userdata:sub(sep + 1))
+	local thoff = par.thoff
+	local proto = skb:getuint8(9)
+
+	local packetdst = skb:getuint32(16)
+	if packetdst ~= dst_ip then
+		return action.ACCEPT
+	end
+
+	if proto == udp then
```

Comment:
```suggestion
	if proto ~= udp then
		goto accept
```

wondering if we really need this condition.. couldn't it be simply handle by the iptables rule?

Reply:
Yeah, the iptables rule part `-p udp --sport 53` should handle it.